Update to Molo version 6

nathanbegbie opened the pull request:
This PR aims to do the following things:
- Update NurseConnect to run on Molo v6 and thus Django 1.11 and Wagtail 1.13
- Update and test NurseConnect to run Python 3
- Small syntax cleanups
- Improve test coverage by updating what code is monitored

nathanbegbie committed:
Update Travis to test with Python3

nathanbegbie committed:
Update requirements to V6 for molo and packages

nathanbegbie committed:
Update to handle Python3 and Django 1.11

- Update use of URL patterns
- small syntax changes
- add wagtail_personalisation to accomodate updated molo.surveys

nathanbegbie committed:
Use `.format` instead of `%` for string substitution

nathanbegbie committed:
Update setup.py file

- ensure that `pip install -e .` works by converting dependencies from `requirements.txt` from a filter into a list
- update meta information
- update how we read in files

nathanbegbie committed:
Fix breaking test by comparing dicts instead of strings

This test broke in Python 3 because we were comparing a string and bytes. See https://stackoverflow.com/questions/6269765/what-does-the-b-character-do-in-front-of-a-string-literal#6273618 for more detail.
It is better practice to compare dicts in this case anyway as string comparison is brittle and cannot identify which values differ when compared.

nathanbegbie committed:
Fix case where user is logged out on password reset

See https://docs.djangoproject.com/en/1.11/topics/auth/default/#session-invalidation-on-password-change
Django invalidates the session's password hash upon password change. We need to update the user session with the new password hash to avoid the user being logged out.

nathanbegbie committed:
Ignore wagtailsearch in test coverage

See related PR for wagtail core https://github.com/wagtail/wagtail/pull/3004/files

miltontony reviewed (approved):
👍 This looks good from my side

rudigiesler commented on nurseconnect/tasks.py:
```diff
@@ -53,8 +53,8 @@ def nurses_registered():
-                "period":
-                    str(datetime.now().year) + "%02d" % datetime.now().month,
+                "period": "{}{:02d}".format(str(datetime.now().year),
+                                            datetime.now().month),
```
I'm thinking we could maybe use `strftime` here instead? (and everywhere else that we're doing this formatting).

We could maybe also store the formatting string for this somewhere central, and then everywhere uses the same formatting string, so that there's one place to change this date format.

rudigiesler reviewed (commented):
Just one little comment from me, the rest looks great otherwise 🍰

alexmuller commented on .travis.yml:
```diff
@@ -57,3 +57,6 @@ jobs:
       after_success: []
+  allow_failures:
+    - python: '3.6'
+      env: TEST=build
```
I don't think this environment variable is required (unless I'm missing something), it was just helpful on molo core to split the CI so that each bit could pass individually.

nathanbegbie replied:
Ah right! I think I just added this temporarily while I fixed up all the python things. I'm actually going to remove `allow_failures` entirely, as everything should now be passing with Python3

alexmuller commented on nurseconnect/forms.py:
```diff
@@ -119,18 +119,18 @@ def __init__(self, *args, **kwargs):
                         "class": "Form-input",
-                        "placeholder": "Enter " + str(question).lower(),
-                        "for": "sq" + str(index)
+                        "placeholder": "Enter ".format(str(question).lower()),
```
I think you're calling `format()` here without anywhere in the string to insert it. I don't really understand why this doesn't make Python raise an error, I feel like it should.

```
>>> 'string'.format(1)
'string'
>>> 'string {0}'.format(1)
'string 1'
```

nathanbegbie replied:
Good catch! Thanks :)

alexmuller commented on nurseconnect/forms.py:
```diff
@@ -361,14 +361,15 @@ def __init__(self, *args, **kwargs):
-                        "for": "sq" + str(index)
+                        "placeholder": "Enter {}".format(
+                            str(question).lower()),
+                        "for": "sq".format(str(index))
```
Same here I think maybe?

nathanbegbie committed:
Refactor metrics date formatting into function

nathanbegbie commented:
Currently blocked by https://github.com/praekelt/molo/pull/525

nathanbegbie committed:
Improve test_services

- use setUp to keep tests DRY
- split out tests that expect None response for clarity
- suppress logging to avoid cluttering test output

nathanbegbie committed:
Update .gitignore to handle pytest output

nathanbegbie committed:
Update molo.core form 6.6.2 to 6.7.0

This solves the migration issues that arose from migrating existing data
